src/dataworker/Dataworker.ts

```diff
@@ -618,14 +618,46 @@ export class Dataworker {
       }
     }
 
-    // Root bundle is valid, attempt to persist it to DA layer if not already there.
+    // Root bundle is valid, attempt to persist the raw bundle data and the merkle leaf data to DA layer
+    // if not already there.
     if (persistBundleData && isDefined(bundleData)) {
-      await persistDataToArweave(
-        this.clients.arweaveClient,
-        bundleData,
-        this.logger,
-        `bundles-${bundleData.bundleBlockRanges}`
-      );
+      await Promise.all([
+        persistDataToArweave(
+          this.clients.arweaveClient,
+          bundleData,
+          this.logger,
+          `bundles-${bundleData.bundleBlockRanges}`
+        ),
+        persistDataToArweave(
+          this.clients.arweaveClient,
+          {
+            poolRebalanceLeaves: expectedTrees.poolRebalanceTree.leaves.map((leaf) => {
+              return {
+                ...leaf,
+                proof: expectedTrees.poolRebalanceTree.tree.getHexProof(leaf),
+              };
+            }),
+            poolRebalanceRoot: expectedTrees.poolRebalanceTree.tree.getHexRoot(),
+            relayerRefundLeaves: expectedTrees.relayerRefundTree.leaves.map((leaf) => {
+              return {
+                ...leaf,
+                proof: expectedTrees.relayerRefundTree.tree.getHexProof(leaf),
+              };
+            }),
+            relayerRefundRoot: expectedTrees.relayerRefundTree.tree.getHexRoot(),
+            slowRelayLeaves: expectedTrees.slowRelayTree.leaves.map((leaf) => {
+              const { relayData, chainId, updatedOutputAmount } = leaf;
+              return {
+                ...leaf,
+                proof: expectedTrees.slowRelayTree.tree.getHexProof({ relayData, chainId, updatedOutputAmount }),
```

Comment:
Is there incidentally something additional in `leaf` that we don't want to persist to Arweave? Per the interface definition it should be possible to get away without this dereferencing.

```suggestion
              return {
                ...leaf,
                proof: expectedTrees.slowRelayTree.tree.getHexProof(leaf),
```